### ISSUE TITLE
Add autoconfigure support

### ISSUE THREAD

**Issue by iltar** [hostnet/form-handler-bundle#46]
Currently I have to add some config in all my applications where I define form handlers:

```yml
services:
    _instanceof:
        Hostnet\Component\Form\FormHandlerInterface:
            tags: [{ name: form.handler }]
            public: true
        Hostnet\Component\FormHandler\HandlerTypeInterface:
            tags: [{ name: form.handler }]
            public: true
```

**Comment by hboomsma** [comment]
@iltar isn't this already fixed by #42? Or should we add more?

**Comment by iltar** [comment]
Currently they still have to be made public
